internal/controller/operatorconfigmap_controller.go

```diff
@@ -950,3 +1001,71 @@ func (c *OperatorConfigMapReconciler) deleteDelegatedCSI() error {
 	}
 	return nil
 }
+
+// TODO: Remove it in 4.20 release
+func (c *OperatorConfigMapReconciler) getCsiCustomTolerations() ([]corev1.Toleration, []corev1.Toleration, error) {
+	const (
+		rookCephOperatorConfigName  = "rook-ceph-operator-config"
+		csiPluginTolerationKey      = "CSI_PLUGIN_TOLERATIONS"
+		csiProvisionerTolerationKey = "CSI_PROVISIONER_TOLERATIONS"
+	)
+	var nodePluginTolerations, controllerPluginTolerations []corev1.Toleration
+
+	// list the rook ceph cm
+	rookCephOperatorConfig := &corev1.ConfigMap{}
+	rookCephOperatorConfig.Name = rookCephOperatorConfigName
+	rookCephOperatorConfig.Namespace = c.OperatorNamespace
+
+	if err := c.get(rookCephOperatorConfig); kerrors.IsNotFound(err) {
+		return nodePluginTolerations, controllerPluginTolerations, nil
+	} else if err != nil {
+		return nodePluginTolerations, controllerPluginTolerations, fmt.Errorf("failed to get Rook Ceph Operator ConfigMap: %v", err)
+	}
+
+	// Get the CSI tolerations from the Rook Ceph Operator ConfigMap
+	csiPluginTolerations := rookCephOperatorConfig.Data[csiPluginTolerationKey]
+	if csiPluginTolerations != "" {
+		var err error
+		if nodePluginTolerations, err = parseTolerations(csiPluginTolerations); err != nil {
+			return nodePluginTolerations, controllerPluginTolerations, fmt.Errorf("failed to parse csi plugin tolerations: %v", err)
+		}
+	}
+
+	csiProvisionerToleration := rookCephOperatorConfig.Data[csiProvisionerTolerationKey]
+	if csiProvisionerToleration != "" {
+		var err error
+		if controllerPluginTolerations, err = parseTolerations(csiProvisionerToleration); err != nil {
+			return nodePluginTolerations, controllerPluginTolerations, fmt.Errorf("failed to parse csi controllerPlugin tolerations: %v", err)
```

Comment:
You shouldn't return the vars because at this point you will have one list filled in and one not based on the code order
```suggestion
			return nil, nil, fmt.Errorf("failed to parse csi controllerPlugin tolerations: %v", err)
```